G1ANT.Addon.Net/Structures/AttachmentStructure.cs

```
@@ -0,0 +1,122 @@
+            tempFileService = iTempFileService ?? new TempFileService();
+            getMd5HashService = iGetMd5HashService ?? new GetMd5HashService();
+        }
+
+        public override Structure Get(string index = "")
+        {
+            if (string.IsNullOrWhiteSpace(index))
+            {
+                throw new NotSupportedException("Index not found in AttachmentStructure");
+            }
+
+            switch (index.ToLower())
+            {
+                case NameIndex:
+                    return new TextStructure(Value.Name, null, Scripter);
+                case SizeIndex:
+                    return new IntegerStructure(Value.Size, null, Scripter);
+                case TypeIndex:
+                    return new TextStructure(Value.Type, null, Scripter);
+                case PathIndex:
+                    pathToTempFile = SaveAndGetPath(Value.MimeEntity, Value.Name);
+                    return new PathStructure(pathToTempFile, null, Scripter);
+                default:
+                    throw new ArgumentException($"Unknown index '{index}'", nameof(index));
+            }
+
+        }
+
+        private string SaveAndGetPath(MimeEntity mimeEntity, string fileName)
+        {
+            var attachmentNameHash = GetAttachmentNameHash(fileName, mimeEntity.ContentId);
+            var filePath = GetAttachmentTempFileNamePath($"{attachmentNameHash}.{fileName}");
+            if (!File.Exists(filePath))
+            {
+                if (mimeEntity is MessagePart msgPart)
+                {
+                    using (var stream = File.Create(filePath))
+                        msgPart.Message.WriteTo(stream);
+                }
+                else if (mimeEntity is MimePart mimePart)
+                {
+                    using (var stream = File.Create(filePath))
+                        mimePart.Content.DecodeTo(stream);
+                }
+                else
+                    return string.Empty;
```

Should throw an exception